x-pack/platform/plugins/shared/task_manager/server/task_store.ts

```diff
@@ -455,6 +602,27 @@ export class TaskStore {
    * @returns {Promise<SavedObjectsBulkDeleteResponse>}
    */
   public async bulkRemove(taskIds: string[]): Promise<SavedObjectsBulkDeleteResponse> {
+    const taskInstances = await this.bulkGet(taskIds);
+    const apiKeyIdsToRemove: string[] = [];
+
+    taskInstances.forEach((taskInstance) => {
+      const unwrappedTaskInstance = unwrap(taskInstance) as ConcreteTaskInstance;
+      if (
+        unwrappedTaskInstance.userScope?.apiKey &&
+        !unwrappedTaskInstance.userScope?.apiKeyCreatedByUser
+      ) {
+        apiKeyIdsToRemove.push(
+          Buffer.from(unwrappedTaskInstance.userScope.apiKey, 'base64').toString().split(':')[0]
+        );
+      }
+    });
+
+    if (apiKeyIdsToRemove.length) {
+      this.security.authc.apiKeys.invalidateAsInternalUser({
```

Comment:
looks like this is missing an await? should we catch any errors that might be. thrown from this?